src/main/java/com/faforever/client/main/event/DeleteLocalReplayEvent.java

```diff
@@ -0,0 +1,12 @@
+package com.faforever.client.main.event;
+
+import javafx.event.Event;
+import javafx.event.EventType;
+
+public class DeleteLocalReplayEvent extends Event {
+  public static final EventType<DeleteLocalReplayEvent> DELETE_LOCAL_REPLAY_EVENT_TYPE = new EventType<>(ANY);
+
+  public DeleteLocalReplayEvent() {
+    super(DELETE_LOCAL_REPLAY_EVENT_TYPE);
+  }
+}
```

Comment:
For events we actually use the Google EventBus if you search for it in the source you should see a few examples. This allows arbitrary objects to be events

Reply:
It seems I need to change a several classes constructors to have access to that EventBus. Does that sound correct or is there a better way to inject this dependency into LocalReplayVaultController? Any hint appreciated :)

Reply:
@Sheikah45 when you get a chance, not sure if you saw my reply buried in here

Reply:
Yes you should add the eventbus to the constructor